cmd/antrea-agent/options.go

```diff
@@ -403,11 +409,17 @@ func (o *Options) setK8sNodeDefaultOptions() {
 		}
 	}
 
-	if features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.Enable {
-		// Multicluster.Enable is deprecated but it may be set by an earlier version
-		// deployment manifest. If it is set to true, pass the value to
-		// Multicluster.EnableGateway.
-		o.config.Multicluster.EnableGateway = true
+	if features.DefaultFeatureGate.Enabled(features.Multicluster) {
+		if o.config.Multicluster.Enable {
+			// Multicluster.Enable is deprecated but it may be set by an earlier version
+			// deployment manifest. If it is set to true, pass the value to
+			// Multicluster.EnableGateway.
+			o.config.Multicluster.EnableGateway = true
+		}
+
+		if o.config.Multicluster.EnableGateway && o.config.Multicluster.Namespace == "" {
```

Comment:
This should be done in `setK8sNodeDefaultOptions()`. If that is simpler, we can move the whole block (line 412 to 422) to setK8sNodeDefaultOptions().

Reply:
Oh, you are right. I mistook it to be Validate(). Sorry.